challenges.yml

```diff
@@ -875,3 +875,153 @@ extending_v_for_key:
           as it's not duplicated. Vue will work its magic under the hood so that
           items get rendered correctly
       correct: b
+
+extending_prop_trap:
+  question: |
+    Imagine you have this component that renders the name of a Pizza object,
+    which you can in turn modify by typing text into an input element:
+
+    ```vue
+    <template>
+        <label>Name:</table>
+        <pizza-name-input :pizza="pizza" />
+    </template>
+
+    <script>
+    import { getPizza } from '@/services/get-pizza';
+    import PizzaNameInput from '@/components/pizza-name-input';
+
+    export default {
+        name: 'PizzaFormField',
+        components: {
+            PizzaNameInput,
+        },
+        data() {
+            pizza: null,
+        },
+        async created() {
+            this.pizza = (await getPizza()).data['hydra:member'];
+        },
+    };
+    </script>
+    ```
+
+    ```vue
+    <template>
+        <input type="text" v-model="pizza.name" />
+    </template>
+
+    <script>
+    export default {
+        name: 'PizzaNameInput',
+        props: {
+            pizza: {
+                type: Object,
+                required: true,
+            },
+        },
+    };
+    </script>
+    ```
+
+    What is the mistake we've just made here?
+  explanation: |
+    Our code will *still* work in this situation. But if ever we decide to modify
+    the object in the parent component, our changes will be wiped out and be lost.
+    Also, the value of the `pizza.name` in our parent object will *not* change
+    when we ourselves change it down the child's prop.
```

Comment:
Again, I think this is not true It (maybe) will not cause the parent component to re-render, but I think it would actually change `pizza.name` in the parent component's data.